### PR TITLE
Fix/TreasuryBalances

### PR DESCRIPTION
**Pull request by jasmeets-6575**
## Summary by CodeRabbit

- **New Features**
	- Introduced four additional treasury addresses for enhanced treasury management.
	- Updated the balance-fetching logic to support multiple treasury addresses, improving asset balance retrieval.
	- Added a new modal component to display detailed treasury information, enhancing user interaction and experience.
	- Expanded asset enumeration to include a new asset type, `MYTH`.
	- Implemented a new API endpoint for retrieving Myth token balances.
	- Introduced a custom React hook for managing Myth token balance fetching.

- **Bug Fixes**
	- Improved error handling during the balance-fetching process.

- **Documentation**
	- Updated the `ChainProps` interface to include new optional properties for the additional treasury addresses.

### CODE REVIEW

File: src/components/Home/overviewData/LatestTreasuryOverview.tsx [hunk 1]
```diff
@@ -55,6 +58,13 @@ const LatestTreasuryOverview = ({ currentTokenPrice, available, priceWeeklyChang
 		)
 	);
 
+	const totalInUSD =
+		Number(available.value) * Number(currentTokenPrice.value) +
```
[the remark this answers is not in the document]
Comment (jasmeets-6575) [reply src/components/Home/overviewData/LatestTreasuryOverview.tsx]: they are already in BN these values are coming from props where they are handled in bn

File: src/components/Home/overviewData/LatestTreasuryOverview.tsx [hunk 2]
```diff
@@ -107,6 +99,15 @@ const LatestTreasuryOverview = ({ currentTokenPrice, available, priceWeeklyChang
 		}
 	}, [assethubApiReady, hydrationApiReady, fetchAssetsAmount, fetchHydrationAssetsAmount]);
 
+	const closeModal = () => setIsModalVisible(false);
+
+	const dotPrice = 29_600_000 * Number(currentTokenPrice.value);
```
Comment (alphainfinitus) [inline src/components/Home/overviewData/LatestTreasuryOverview.tsx:104]: use BN please and don't use hard-coded prices

**Review by alphainfinitus — CHANGES_REQUESTED** [review]

File: src/components/Home/overviewData/LatestTreasuryOverview.tsx [hunk 3]
```diff
@@ -107,6 +99,18 @@ const LatestTreasuryOverview = ({ currentTokenPrice, available, priceWeeklyChang
 		}
 	}, [assethubApiReady, hydrationApiReady, fetchAssetsAmount, fetchHydrationAssetsAmount]);
 
+	const closeModal = () => setIsModalVisible(false);
+
+	const totalUsdcPrice = formatUSDWithUnits(new BN(assethubValues.usdcValue).add(new BN(hydrationValues.usdcValue)).div(BN_MILLION).add(new BN(3000000)).toString());
```
Comment (alphainfinitus) [inline src/components/Home/overviewData/LatestTreasuryOverview.tsx:104]: please assign these magic numbers to constants

File: src/components/Home/overviewData/LatestTreasuryOverview.tsx [hunk 4]
```diff
@@ -107,6 +99,18 @@ const LatestTreasuryOverview = ({ currentTokenPrice, available, priceWeeklyChang
...
+	const totalUsdtPrice = formatUSDWithUnits(new BN(assethubValues.usdtValue).add(new BN(hydrationValues.usdtValue)).div(BN_MILLION).add(new BN(2420000)).toString());
+
+	const dotPrice = new BN('29600000').mul(new BN(String(Number(currentTokenPrice.value || 0) * 1000000))).div(new BN('1000000'));
```
Comment (alphainfinitus) [inline src/components/Home/overviewData/LatestTreasuryOverview.tsx:107]: magic numbers here as well

**Review by alphainfinitus — CHANGES_REQUESTED** [review]

File: pages/api/v1/treasury-amount-history/mythTokenAmount.ts [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+		return { error: error instanceof Error ? error.message : 'Data Not Available' };
+	}
+};
+
+const handler = async (req: NextApiRequest, res: NextApiResponse<IMythTokenBalanceResponse>): Promise<void> => {
+	storeApiKeyUsage(req);
+
+	const network = req.headers['x-network'] as string;
+	const address = chainProperties[network]?.assetHubTreasuryAddress4;
+	const apiUrl = `${chainProperties[network]?.assethubExternalLinks}/api/scan/foreignAssets/account/balances`;
```
Comment (alphainfinitus) [inline pages/api/v1/treasury-amount-history/mythTokenAmount.ts:58]: this will result in `undefined/api/....` 
Please add network checks

**Review by alphainfinitus — CHANGES_REQUESTED** [review]

**Pull request merged**